test/codeu/chat/server/JSONTest.java

```diff
@@ -0,0 +1,204 @@
+        JSON log = new JSON();
+        Model model = new Model();
+        Model comparer = new Model();
+        comparer.add(new User(createUuid("1.1074501217"), "Krager", Time.fromMs(4098590)));
+        comparer.add(new User(createUuid("1.3566231147"), "FakeUserName1234", Time.fromMs(48574)));
+        comparer.add(new ConversationHeader(createUuid("1.3959833157"), createUuid("1.1074501217"),  Time.fromMs(98094), "Chat1"));
+        comparer.add(new ConversationHeader(createUuid("1.3959833157"), createUuid("1.1074501217"),  Time.fromMs(59485), "Chat2"));
+        comparer.add(new Message(createUuid("1.3928689216"), Uuid.NULL, Uuid.NULL, Time.fromMs(49589), createUuid("1.1074501217"),   "I'm so alone"));
+        comparer.add(new Message(createUuid("1.2016074193"), Uuid.NULL, Uuid.NULL, Time.fromMs(9485), createUuid("1.1074501217"),   "I need friends"));
+        comparer.add(new Message(createUuid("1.384992272"), Uuid.NULL, Uuid.NULL, Time.fromMs(5098), createUuid("1.1074501217"),   "Julia'sFixesAreNotImplementedYetCanYouTell?"));
+
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input4.json");
+            // The path can vary from device, so this may not compile if it's not the correct path for your project
+            // set up. A suggested path is as follows
+            //model = log.readFromFile("CodeU/test/codeu/chat/server/input4.json");
+        } catch (IOException e) {
+            System.out.println("File not handled properly");
+        }
+        assertTrue(
+                "Check that the user has the correct id", true);
+        assertEquals(
+                compare(model.userById().all().iterator(), comparer.userById().all().iterator()), true);
+        assertEquals(
+                compare(model.userByText().all().iterator(), comparer.userByText().all().iterator()), true);
+        assertEquals(
+                compare(model.userByTime().all().iterator(), comparer.userByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationById().all().iterator(), comparer.conversationById().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByTime().all().iterator(), comparer.conversationByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByText().all().iterator(), comparer.conversationByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageById().all().iterator(), comparer.messageById().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByText().all().iterator(), comparer.messageByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByTime().all().iterator(), comparer.messageByTime().all().iterator()), true);
+    }
+
```

Comment:
I would maybe expect to see tests for bad cases as well. What happens if the JSON contains invalid data? Or other corner cases? Tests should make sure your code works in corner cases, not just in cases you expect.